tests/test_cli.py

```diff
@@ -477,6 +477,35 @@ def test_eval_param(tmpdir):
     assert float(recall.group(1)) == 0.0
 
 
+def test_eval_resultsfile(tmpdir):
+    tmpdir.join('doc1.txt').write('doc1')
+    tmpdir.join('doc1.key').write('dummy')
+    tmpdir.join('doc2.txt').write('doc2')
+    tmpdir.join('doc2.key').write('none')
+    tmpdir.join('doc3.txt').write('doc3')
+    result = runner.invoke(
+        annif.cli.cli, [
+            'eval', '--results-file', 'test_file.txt', 'dummy-en',
+            str(tmpdir)])
+    assert not result.exception
+    assert result.exit_code == 0
```

Comment:
I extended the test to cover these items. It may have grown a bit too large, but since it's all testing the contents of the same output file I kept it like this. 
I hope it's ok that the test is in test_cli rather than test_eval. I realised this might not be ideal only after I'd finished writing it, and moving it around wasn't trivial.